src/constructs/lambda/lambda.ts

```diff
@@ -125,23 +127,55 @@ export class GuLambdaFunction extends Function {
       withoutFilePrefix = false,
       withoutArtifactUpload = false,
       logFormat = LoggingFormat.JSON,
-    } = props;
 
-    const bucketName = bucketNamePath
-      ? StringParameter.fromStringParameterName(scope, "bucketoverride", bucketNamePath).stringValue
-      : GuDistributionBucketParameter.getInstance(scope).valueAsString;
+      /*
+      This default has a consequence that, within a single CFN stack,
+      each lambda function must have a unique Stage/Stack/App triple.
+      Else we'll create the same lambda twice, which is not possible.
+      This error will also be first seen at runtime, not at CDK synth time,
+      i.e. it has a long feedback loop.
+       */
+      functionName = [stage, stack, app].join("-"),
```

Comment:
What's the reason for needing to do this? I haven't been able to discern what you mean by `and avoid cyclical dependencies`.

As you say, the uniqueness of these isn't a given and I think this is a bit surprising. It also might create problems in the future if we need to do a replacing update for this resource?

Reply:
The log group's name is created using the lambda function's name. An auto-generated lambda name is known at deploy time. Without hard-coding the lambda's name we'll have a cyclical dependency where the lambda requires the log group to exist, and the log group requires the lambda to exist.

The "Generate a random log name" alternative in the PR description would here here...